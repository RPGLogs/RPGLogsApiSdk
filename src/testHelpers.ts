import { buildSdk } from './main';
import btoa from 'btoa';

export function buildTestSdk(): ReturnType<typeof buildSdk> {
  return buildSdk(<string>process.env.TEST_ACCESS_TOKEN);
}

export async function getAccessToken(
  clientId: string,
  clientSecret: string,
  game = 'warcraft'
): Promise<string> {
  if (game === 'warcraft') game = 'www.warcraft';

  const authHeader = 'Basic ' + btoa(clientId + ':' + clientSecret);

  const response = await fetch(`https://${game}logs.com/oauth/token`, {
    method: 'POST',
    headers: {
      Authorization: authHeader,
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: 'grant_type=client_credentials',
  });

  const json = await response.json();

  if (response.status === 200) {
    return json.access_token;
  } else {
    throw new Error('Response was not OK: ' + JSON.stringify(json ?? {}));
  }
}
