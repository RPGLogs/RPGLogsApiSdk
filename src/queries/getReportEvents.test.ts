import { buildTestSdk } from '../testHelpers';

test('returns data', async () => {
  const response = await buildTestSdk().getReportEvents({
    code: 'cFZ274xvNgTbLXCj',
    encounterId: 2824,
    filterExpression: 'ability.name in ("Lava Geysers")',
    endTime: 2147483647,
    includeResources: true,
  });

  expect(response?.reportData?.report).toMatchSnapshot();
});
