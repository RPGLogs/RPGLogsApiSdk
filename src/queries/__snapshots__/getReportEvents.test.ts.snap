// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns data 1`] = `
{
  "code": "cFZ274xvNgTbLXCj",
  "events": {
    "data": [
      {
        "abilityGameID": 422691,
        "fight": 26,
        "sourceID": 135,
        "targetID": -1,
        "timestamp": 4247776,
        "type": "begincast",
      },
      {
        "abilityGameID": 422691,
        "absorb": 0,
        "armor": 5024,
        "attackPower": 0,
        "classResources": [
          {
            "amount": 43,
            "max": 100,
            "type": 3,
          },
        ],
        "facing": -586,
        "fight": 26,
        "hitPoints": 1839754742,
        "itemLevel": 73,
        "mapID": 2233,
        "maxHitPoints": 2037296917,
        "resourceActor": 1,
        "sourceID": 135,
        "spellPower": 0,
        "targetID": -1,
        "timestamp": 4249764,
        "type": "cast",
        "x": -1311624,
        "y": 403040,
      },
      {
        "abilityGameID": 422691,
        "fight": 26,
        "sourceID": 135,
        "targetID": -1,
        "timestamp": 4272723,
        "type": "begincast",
      },
      {
        "abilityGameID": 422691,
        "absorb": 0,
        "armor": 5024,
        "attackPower": 0,
        "classResources": [
          {
            "amount": 83,
            "max": 100,
            "type": 3,
          },
        ],
        "facing": -283,
        "fight": 26,
        "hitPoints": 1678554560,
        "itemLevel": 73,
        "mapID": 2233,
        "maxHitPoints": 2037296917,
        "resourceActor": 1,
        "sourceID": 135,
        "spellPower": 0,
        "targetID": -1,
        "timestamp": 4274724,
        "type": "cast",
        "x": -1313361,
        "y": 403347,
      },
      {
        "abilityGameID": 422823,
        "absorbed": 660556,
        "amount": 0,
        "fight": 26,
        "hitType": 1,
        "mitigated": 99285,
        "sourceID": 135,
        "targetID": 3,
        "timestamp": 4277777,
        "type": "damage",
        "unmitigatedAmount": 759841,
      },
      {
        "abilityGameID": 422691,
        "fight": 26,
        "sourceID": 135,
        "targetID": -1,
        "timestamp": 4346357,
        "type": "begincast",
      },
      {
        "abilityGameID": 422691,
        "absorb": 0,
        "armor": 5024,
        "attackPower": 0,
        "classResources": [
          {
            "amount": 43,
            "max": 100,
            "type": 3,
          },
        ],
        "facing": -295,
        "fight": 26,
        "hitPoints": 1154487987,
        "itemLevel": 73,
        "mapID": 2233,
        "maxHitPoints": 2037296917,
        "resourceActor": 1,
        "sourceID": 135,
        "spellPower": 0,
        "targetID": -1,
        "timestamp": 4348363,
        "type": "cast",
        "x": -1314935,
        "y": 406635,
      },
      {
        "abilityGameID": 422691,
        "fight": 26,
        "sourceID": 135,
        "targetID": -1,
        "timestamp": 4371330,
        "type": "begincast",
      },
      {
        "abilityGameID": 422691,
        "absorb": 0,
        "armor": 5024,
        "attackPower": 0,
        "classResources": [
          {
            "amount": 85,
            "max": 100,
            "type": 3,
          },
        ],
        "facing": -236,
        "fight": 26,
        "hitPoints": 1026041707,
        "itemLevel": 73,
        "mapID": 2233,
        "maxHitPoints": 2037296917,
        "resourceActor": 1,
        "sourceID": 135,
        "spellPower": 0,
        "targetID": -1,
        "timestamp": 4373325,
        "type": "cast",
        "x": -1314114,
        "y": 409084,
      },
      {
        "abilityGameID": 422823,
        "absorb": 0,
        "absorbed": 515414,
        "amount": 74962,
        "armor": 58996,
        "attackPower": 25766,
        "classResources": [
          {
            "amount": 67,
            "max": 120,
            "type": 17,
          },
        ],
        "facing": -635,
        "fight": 26,
        "hitPoints": 2759765,
        "hitType": 1,
        "itemLevel": 528,
        "mapID": 2233,
        "maxHitPoints": 2834727,
        "mitigated": 269965,
        "resourceActor": 2,
        "sourceID": 135,
        "spellPower": 1492,
        "targetID": 21,
        "timestamp": 4376846,
        "type": "damage",
        "unmitigatedAmount": 860341,
        "x": -1314484,
        "y": 410114,
      },
      {
        "abilityGameID": 422691,
        "fight": 26,
        "sourceID": 135,
        "targetID": -1,
        "timestamp": 4444732,
        "type": "begincast",
      },
      {
        "abilityGameID": 422691,
        "absorb": 0,
        "armor": 5024,
        "attackPower": 0,
        "classResources": [
          {
            "amount": 44,
            "max": 100,
            "type": 3,
          },
        ],
        "facing": -208,
        "fight": 26,
        "hitPoints": 537684690,
        "itemLevel": 73,
        "mapID": 2233,
        "maxHitPoints": 2037296917,
        "resourceActor": 1,
        "sourceID": 135,
        "spellPower": 0,
        "targetID": -1,
        "timestamp": 4446732,
        "type": "cast",
        "x": -1310398,
        "y": 409370,
      },
      {
        "abilityGameID": 422691,
        "amount": 0,
        "fight": 26,
        "hitType": 10,
        "sourceID": 135,
        "targetID": 2,
        "timestamp": 4446732,
        "type": "damage",
      },
      {
        "abilityGameID": 422691,
        "fight": 26,
        "sourceID": 135,
        "targetID": -1,
        "timestamp": 4469731,
        "type": "begincast",
      },
      {
        "abilityGameID": 422691,
        "absorb": 0,
        "armor": 5024,
        "attackPower": 0,
        "classResources": [
          {
            "amount": 85,
            "max": 100,
            "type": 3,
          },
        ],
        "facing": -741,
        "fight": 26,
        "hitPoints": 399766979,
        "itemLevel": 73,
        "mapID": 2233,
        "maxHitPoints": 2037296917,
        "resourceActor": 1,
        "sourceID": 135,
        "spellPower": 0,
        "targetID": -1,
        "timestamp": 4471732,
        "type": "cast",
        "x": -1308453,
        "y": 408590,
      },
    ],
    "nextPageTimestamp": null,
  },
}
`;
